bundle/src/main/java/com/adobe/acs/commons/http/headers/impl/PropertyBasedDispatcherMaxAgeHeaderFilter.java

```diff
@@ -112,16 +110,13 @@ protected String getHeaderValue(HttpServletRequest request) {
         return super.getHeaderValue(request);
     }
 
-    private Resource getResource(HttpServletRequest request, ResourceResolver resourceResolver) {
-        PathInfo pathInfo = new PathInfo(request.getRequestURI());
-        Resource reqResource = resourceResolver.getResource(pathInfo.getResourcePath());
-        if (reqResource != null) {
-            if (reqResource.isResourceType("cq:Page")) {
-                return reqResource.getChild(JcrConstants.JCR_CONTENT);
-            }
-            return reqResource;
+    private Resource getResource(SlingHttpServletRequest slingRequest) {
```

Comment:
this method seems to be present in both classes of this PR. Would it make sense to move it to the superclass?

Reply:
done